waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -107,8 +113,11 @@ genPackageJson spec waspDependencies = do
             ]
       )
 
-npmDepsForWasp :: N.NpmDepsForWasp
-npmDepsForWasp =
+-- This takes a `Maybe AppSpec` so it can work in contexts where
+-- we want to display base dependencies that do not rely on optional
+-- features, as in the `deps` CLI command.
+npmDepsForWasp :: Maybe AppSpec -> N.NpmDepsForWasp
```

Comment:
I guess it makes sense, right? It seems we need to know AppSpec in order to figure out the dependencies -> well that does make sense indeed. So I don't see any other way! It is inherently needed! Unless we would equip deps themselves with some kind of predicates that would later be resolved against AppSpec to figure out for each of them if they are needed or not, but that sounds overly complicated at the moment.

Reply:
The only followup question I have is we want this callable by deps in CLI where we don't have an AppSpec, hence the `Maybe AppSpec` part... any way/need to clean that part up?

Reply:
I actually may have an idea to make it `AppSpec` by separating out required and opt-in dependencies. Will test out on Monday 👍🏻

Reply:
While that might be a direction to go, I wonder if trying to obtain AppSpec in CLI is the better way to go, since we have more information all together. Depends on how hard it is do achieve.

Reply:
Yeah AppSpec in CLI is definitely preferred. Didn't think it would be as straightforward as the comment before mentioned. I will look into that today and hope it works out, as anything else is less ideal/hacky :D